src/mono/wasm/runtime/jiterpreter.ts

```diff
@@ -0,0 +1,2967 @@
+            }, WasmValtype.i32
+        );
+        builder.defineType(
+            "mathop_d_d", {
+                "value": WasmValtype.f64,
+            }, WasmValtype.f64
+        );
+        builder.defineType(
+            "mathop_dd_d", {
+                "lhs": WasmValtype.f64,
+                "rhs": WasmValtype.f64,
+            }, WasmValtype.f64
+        );
+        builder.defineType(
+            "trace_eip", {
+                "traceId": WasmValtype.i32,
+                "eip": WasmValtype.i32,
+            }, WasmValtype.void
+        );
+        builder.defineType(
+            "newobj_i", {
+                "ppDestination": WasmValtype.i32,
+                "vtable": WasmValtype.i32,
+            }, WasmValtype.i32
+        );
+        builder.defineType(
+            "newarr", {
+                "ppDestination": WasmValtype.i32,
+                "length": WasmValtype.i32,
+                "vtable": WasmValtype.i32,
+            }, WasmValtype.i32
+        );
+        builder.defineType(
+            "localloc", {
+                "destination": WasmValtype.i32,
+                "len": WasmValtype.i32,
+                "frame": WasmValtype.i32,
+            }, WasmValtype.void
+        );
+        builder.defineType(
+            "ld_del_ptr", {
+                "ppDestination": WasmValtype.i32,
+                "ppSource": WasmValtype.i32,
+            }, WasmValtype.void
+        );
+        builder.defineType(
+            "gettype", {
+                "destination": WasmValtype.i32,
+                "source": WasmValtype.i32,
+            }, WasmValtype.i32
+        );
+        builder.defineType(
+            "castclass", {
+                "destination": WasmValtype.i32,
+                "source": WasmValtype.i32,
+                "klass": WasmValtype.i32,
+                "isinst_instr": WasmValtype.i32,
+            }, WasmValtype.i32
+        );
+        builder.defineType(
+            "try_unbox", {
+                "klass": WasmValtype.i32,
+                "destination": WasmValtype.i32,
+                "source": WasmValtype.i32,
+            }, WasmValtype.i32
+        );
+        builder.defineType(
+            "box", {
+                "vtable": WasmValtype.i32,
+                "destination": WasmValtype.i32,
+                "source": WasmValtype.i32,
+            }, WasmValtype.void
+        );
+
+        builder.generateTypeSection();
+
+        // Import section
+        builder.beginSection(2);
+        const traceImports = getTraceImports();
+        builder.appendULeb(1 + traceImports.length);
+
+        // Emit function imports
+        for (let i = 0; i < traceImports.length; i++) {
+            mono_assert(traceImports[i], () => `trace #${i} missing`);
+            const wasmName = compress ? i.toString(16) : undefined;
+            builder.defineImportedFunction("i", traceImports[i][0], traceImports[i][1], wasmName);
+        }
+
+        builder.appendName("i");
```

Comment:
nit: please rename to descriptive names

Reply:
unfortunately, the names end up in the wasm binary, so making them longer pushes us closer to the 4kb limit. They originally had descriptive names and the result was that the module header was over 500 bytes of junk